Source/Core/Core/IOS/Network/Net.cpp

```diff
@@ -625,9 +635,7 @@ IPCCommandResult NetIPTop::IOCtl(const IOCtlRequest& request)
     // Do the level/optname translation
     int nat_level = -1, nat_optname = -1;
 
-    for (auto& map : opt_level_mapping)
-      if (level == map[1])
-        nat_level = map[0];
+    nat_level = mapWiiSockOptToNative(level);
```

Comment:
Please initialize `nat_level` with this value instead of initializing it with -1.